client/landing/gutenboarding/hooks/use-typer.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import { useState, useEffect } from 'react';
+
+/**
+ * A React hook that returns typing-machine animated strings
+ *
+ * @param {Array<string>} texts An array of strings you want to create the typing effect for
+ * @param {boolean?} enabled Whether the animation is enabled
+ * @param {number?} delayBetweenCharacters The typing delay time between two charactors. Default is 75ms
+ * @param {number?} delayBetweenWords The typing delay time between two words. Default is 750ms
+ */
+export default function useTyper(
+	texts: Array< string >,
+	enabled = true,
+	delayBetweenCharacters = 75,
+	delayBetweenWords = 750
+) {
+	const [ indices, setIndices ] = useState( { char: 0, word: 0 } );
```

Comment:
Being able to break state down into units rather than shoving everything into an object is one of the big advantages of `useState` over class component's `setState`. Shall we keep them separate here?

Reply:
I agree but I joined them for a reason. Whenever you update the state a setTimeout will be called because your update will change the deps values.

So in [this line](https://github.com/Automattic/wp-calypso/pull/40431/files#diff-4b30f36c7b54c2125d9a0f7663b3a335R38), I update both indices (char = 0, word++). If the states were separate, the first update will probably trigger an update => set a timeout and mess things up. Having one state means a linear predictable stream of events.